### PR TITLE
Move VERSION to our public API, expose STREAM_STDOUT et.al.

### LINKED ISSUE #915: Highlights refactoring broke "highlight.ERROR/WARNING"
[This refactoring](https://github.com/SublimeLinter/SublimeLinter/commit/f3a57dc) of the `highlight` module broken the [SublimeLinter-annotations](https://github.com/SublimeLinter/SublimeLinter-annotations) linter. I was going to send a pull-request to the other repository but figured it's better to report the issue here since the code is still under development _(for version 4.0.0)_.

```
reloading plugin SublimeLinter-annotations.linter
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 109, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 584, in _check_name_wrapper
  File "<frozen importlib._bootstrap>", line 1022, in load_module
  File "<frozen importlib._bootstrap>", line 1003, in load_module
  File "<frozen importlib._bootstrap>", line 560, in module_for_loader_wrapper
  File "<frozen importlib._bootstrap>", line 868, in _load_module
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "/Users/root/Library/Application Support/Sublime Text 3/Packages/SublimeLinter-annotations/linter.py", line 15, in <module>
    from SublimeLinter.lint import highlight, Linter
ImportError: cannot import name highlight
```

### PR DESCRIPTION
**Pull request by kaste**
Fixes #915

### CODE REVIEW

**Comment by kaste** [comment]
This is the simplest possible commit to move forward and close the issue. 

Of course we need a commit for SL3, which moves VERSION to lint.__init__.py as well

File: __init__.py [hunk 1]
```diff
@@ -1,2 +1,5 @@
-"""Provides an API version for plugins."""
-VERSION = 4
+"""Hi.
```
Comment (braver) [inline __init__.py]: Shouldn’t we simply drop this file?

**Comment by braver** [comment]
Also, is this really necessary? Does it cause the double loading even if the version is checked like `getattr(SublimeLinter, 'VERSION', 3)`? It really sucks because we already have plugins that are using this.

**Comment by braver** [comment]
Ok, nevermind, in favour of merging this. Then move on to fix #899 to clean up the rest. I don't mind having to go through the plugins to align them again.

@kaste feel free to merge, but perhaps explain why we keep the __init__ at the top level?

**Comment by kaste** [comment]
There is no reason for `__init__.py`.  It would be *necessary* if we get unittests, and maybe some code analyzers need it.  It was there so I didn't remove it. Then usually python devs look for these files, and we could eventually add some documentation in there, roughly explaining how this all works.

**Comment by braver** [comment]
Ok, sounds good. Kinda like a common/conventional entry point.

**Comment by FichteFoll** [comment]
FWIW, `__init__.py` in the root is *always* loaded twice because, as explained in the referenced LSP issue, it is loaded by Python before any submodule and then again by ST itself. It shouldn't matter much if the work it does isn't significant (i.e. just import stuff from elsewhere or some constant definitions), but should be considered for larger things.

**Pull request merged**